### ISSUE TITLE
Inconsistent IDs for FMN and FMNH2

### ISSUE THREAD

**Issue by Devlin-Moyer** [SysBioChalmers/Human-GEM#711]
#### Current behavior:
I was starting to work on implementing the changes I proposed in #609, and I noticed inconsistencies in the metabolite IDs for FMN and FMNH2 metabolites in different compartments:

ID | Name | Compartment | Formula | Charge
---|---|---|---|---
`MAM01828c` | FMN | c | C17H**19**N4O9P | -2
`MAM01828x` | FMN | x | C17H**19**N4O9P | -2
`MAM01828e` | FMN | e | C17H**19**N4O9P | -2
`MAM20020r` | FMN | r | C17H**18**N4O9P | -2
`MAM20023i` | FMN | i | C17H**18**N4O9P | -2
`MAM20019r` | FMNH2 | r | C17H**21**N4O9P | -2
`MAM20022i` | FMNH2 | i | C17H**21**N4O9P | -2

(These are all of the metabolites currently in Human-GEM that have "FMN" in their names. Also I bolded the number of hydrogens in the empricial formulas just to make it clear that that's the only part that changes)

#### Expected feature/value/output:

- [X] All of the FMN metabolites should have the same IDs except for the compartment suffix, so `MAM20020r` should be `MAM01828r` and `MAM20023i` should be `MAM01828i`.
- [X] Same for the FMNH2 metabolites: `MAM20022i` should be `MAM20019i`.
- [X] The empirical formulas for all FMN metabolites should be the same. According to [ChEBI](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=17621), [PubChem](https://pubchem.ncbi.nlm.nih.gov/compound/643976), and  [KEGG](https://www.genome.jp/entry/C00061), the formula for (neutral) FMN should be `C17H21N4O9P`, and since these represent the -2 ion of FMN, the formulas for `MAM20020r` and `MAM20023i` should be changed to `C17H19N4O9P`.
- [X] All of the FMN metabolites should be associated with the same KEGG ID, so `MAM20020r` and `MAM20023i` should both be associated with [C00061](https://www.genome.jp/entry/C00061).
- [X] Relatedly, neither of the FMNH2 metabolites are currently associated with a KEGG ID, so they can both be associated with [C01847](https://www.genome.jp/entry/C01847).

**Comment by haowang-bioinfo** [comment]
thanks for the scrutiny, go ahead to implementation please

**Comment by Devlin-Moyer** [comment]
FYI I just edited my proposed changes a bit because I realized the empirical formulas I looked up were for _neutral_ FMN and FMNH2, but these metabolites represent their -2 ions since their phosphate groups deprotonate in water at physiological pHs.

**Comment by haowang-bioinfo** [comment]
yes, KEGG often picks neutral state for metabolites - in fact the protonation state under pH 7.4 is favoured

**Comment by haowang-bioinfo** [comment]
let's keep this issue open until the charge-balancing fixed

**Comment by edkerk** [comment]
Thanks. Most of this is already resolved on `develop`: the FMN metabolites all share `MAM01828*` (formula C17H19N4O9P) and the FMNH2 metabolites all share `MAM20019*` (C17H21N4O9P), so the ID and formula items are done. Two annotation fixes remained.

**KEGG IDs were misfiled in the BiGG column.** They are moved to the KEGG field, with the BiGG IDs corrected to match FAD/FADH2:

| Metabolite | KEGG | BiGG |
|---|---|---|
| FMN, all compartments | [C00061](https://www.kegg.jp/entry/C00061) | fmn |
| FMNH2, all compartments | [C01847](https://www.kegg.jp/entry/C01847) | fmnh2 |

**FMN cross-references differed between compartments** (c/x/e used generic IDs, r/i used charge-specific ones). All five FMN compartments are now harmonised to the generic forms that c/x/e and the analogous FAD already use:

| | ChEBI | MetaNetX | HMDB | PubChem |
|---|---|---|---|---|
| FMN, all compartments | [CHEBI:17621](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:17621) | MNXM119 | HMDB0001520 | 643976 |

The charge-specific ChEBI that r/i carried ([CHEBI:58210](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:58210)) is FMN(3−), which does not match the model's −2 FMN, and the model's convention is generic anyway (e.g. FAD uses neutral [CHEBI:16238](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:16238), and it uses canonical MetaNetX IDs for 5454 metabolites versus 43 specific ones).

One residual, if you want to chase it: FMNH2 is internally consistent but still uses a non-canonical MetaNetX ID (`MNXM1107623`, one of those 43) and `CHEBI:57618`, mirroring what FMN r/i had. It could be aligned with FADH2 (`MNXM38`) in a follow-up.
